src/Core/Services/Implementations/AzureQueueMailService.cs

```diff
@@ -3,30 +3,71 @@
 using System.Linq;
 using System.Threading.Tasks;
 using Azure.Storage.Queues;
+using Bit.Core.Models.Data;
 using Bit.Core.Models.Mail;
 using Bit.Core.Settings;
+using Microsoft.Azure.Storage;
+using Microsoft.Azure.Storage.Blob;
 using Newtonsoft.Json;
 
 namespace Bit.Core.Services
 {
-    public class AzureQueueMailService : AzureQueueService<IMailQueueMessage>, IMailEnqueuingService
+    public class AzureQueueMailService : IMailEnqueuingService
     {
-        protected override QueueClient QueueClient { get; }
-        protected override JsonSerializerSettings JsonSettings { get; } = new JsonSerializerSettings
+        public const string QueueMessageContainerName = "queue-messages";
+        private readonly QueueClient _queueClient;
+        private readonly CloudBlobClient _blobClient;
+        private readonly JsonSerializerSettings _jsonSettings = new JsonSerializerSettings
         {
             NullValueHandling = NullValueHandling.Ignore
         };
+        private  CloudBlobContainer _queueMessageContainer;
 
         public AzureQueueMailService(
             GlobalSettings globalSettings)
         {
-            QueueClient = new QueueClient(globalSettings.Mail.ConnectionString, "mail");
+            _queueClient = new QueueClient(globalSettings.Mail.ConnectionString, "mail");
+            var storageAccount = CloudStorageAccount.Parse(globalSettings.Mail.ConnectionString);
+            _blobClient = storageAccount.CreateCloudBlobClient();
         }
 
-        public Task EnqueueAsync(IMailQueueMessage message, Func<IMailQueueMessage, Task> fallback) =>
-            CreateAsync(message);
+        public async Task EnqueueAsync(IMailQueueMessage message, Func<IMailQueueMessage, Task> fallback)
+        {
+            await SendMessageAsync(new AzureQueueMessage<IMailQueueMessage>
+            {
+                Message = message
+            });
+        }
+
+        public async Task EnqueueManyAsync(IEnumerable<IMailQueueMessage> messages, Func<IMailQueueMessage, Task> fallback)
+        {
+            await SendMessageAsync(new AzureQueueMessage<IMailQueueMessage>
+            {
+                Messages = messages
+            });
+        }
 
-        public Task EnqueueManyAsync(IEnumerable<IMailQueueMessage> messages, Func<IMailQueueMessage, Task> fallback) =>
-            CreateManyAsync(messages);
+        private async Task SendMessageAsync(AzureQueueMessage<IMailQueueMessage> message)
+        {
+            var json = JsonConvert.SerializeObject(message, _jsonSettings);
+            if (json.Length > _queueClient.MessageMaxBytes)
+            {
+                await InitAsync();
+                var blob = _queueMessageContainer.GetBlockBlobReference($"{message.MessageId}");
+                await blob.UploadTextAsync(json);
+
+                json = JsonConvert.SerializeObject(message.ToBlobBackedMessage(), _jsonSettings);
```

Comment:
This seems kind of weird. Not sure I am following.

Reply:
Here's the logic:
* Serialize the full message
* If the message is > the maximum size for a single message (64KB), write the full message to a blob
* Replace the original full message with a truncated message that just lists the blob name
* Write the truncated message to the queue

When the queue reader pulls down the message, it'll see the message is a stub pointing to a blob, download the blob and process the blob instead of the original queue message